### PR TITLE
[webserver] Compatibility with libmicrohttpd 0.9.71

### PR DESCRIPTION
**Pull request by candrews**
From the libmicrohttpd 0.9.71 release notes:

The release introduces an 'enum MHD_Result' instead of
for certain API misuse bugs by providing better types (not everything is
an 'int').  While this does NOT change the binary API, this change
_will_ cause compiler warnings for all legacy code -- until 'int' is
replaced with 'enum MHD_Result'.

## Motivation and Context

Kodi currently fails to build against libmicrohttpd 0.9.71, failing with this error:
```
/var/tmp/portage/media-tv/kodi-9999/work/kodi-9999/xbmc/network/WebServer.cpp:513:106: error: invalid conversion from ‘int (*)(void*, MHD_ValueKind, const char*, const char*, const char*, const char*, const char*, uint64_t, size_t)’ {aka ‘int (*)(void*, MHD_ValueKind, const char*, const char*, const char*, const char*, const char*, long unsigned int, long unsigned int)’} to ‘MHD_PostDataIterator’ {aka ‘MHD_Result (*)(void*, MHD_ValueKind, const char*, const char*, const char*, const char*, const char*, long unsigned int, long unsigned int)’} [-fpermissive]
  513 |   connectionHandler->postprocessor = MHD_create_post_processor(request.connection, MAX_POST_BUFFER_SIZE, &CWebServer::HandlePostField, static_cast<void*>(connectionHandler));
      |                                                                                                          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |                                                                                                          |
      |                                                                                                          int (*)(void*, MHD_ValueKind, const char*, const char*, const char*, const char*, const char*, uint64_t, size_t) {aka int (*)(void*, MHD_ValueKind, const char*, const char*, const char*, const char*, const char*, long unsigned int, long unsigned int)}
```
[Full build log](https://bugs.gentoo.org/attachment.cgi?id=647328)

Here's the release announcement from libmicrohttpd 0.9.71: https://lists.gnu.org/archive/html/libmicrohttpd/2020-06/msg00013.html
I believe this to be the significant part:
> Furthermore, the release introduces an 'enum MHD_Result' instead of
#defines for MHD_YES/MHD_NO. This is intended to make it easier to check
for certain API misuse bugs by providing better types (not everything is
an 'int').  While this does NOT change the binary API, this change
_will_ cause compiler warnings for all legacy code -- until 'int' is
replaced with 'enum MHD_Result'.
> 
> If you want your code to build without warnings on both older and newer
MHD releases, you may want to introduce a MHD_RESULT as done here:
> 
> https://git.gnunet.org/gnunet.git/tree/src/include/gnunet_mhd_compat.h

## How Has This Been Tested?

Built on my system and the build succeed and Kodi works.
## Screenshots (if appropriate):
n/a
## Types of change

- [X] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [X] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [X] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [X] All new and existing tests passed

### CODE REVIEW

File: xbmc/network/WebServer.cpp [hunk 1]
```diff
@@ -113,8 +113,11 @@ int CWebServer::AskForAuthentication(const HTTPRequest& request) const
 
   LogResponse(request, MHD_HTTP_UNAUTHORIZED);
 
+  // This MHD_RESULT cast is only necessary for libmicrohttpd 0.9.71
+  // The return type of MHD_queue_basic_auth_fail_response was fixed for future versions
+  // See https://git.gnunet.org/libmicrohttpd.git/commit/?id=860b42e9180da4dcd7e8690a3fcdb4e37e5772c5
```
Comment (Rechi) [inline xbmc/network/WebServer.cpp:118]: > The `ColumnLimit` in `.clang-format` is set to `100` which defines line length [...]

https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md#3-formatting

File: xbmc/network/WebServer.cpp [hunk 2]
```diff
@@ -113,8 +113,11 @@ int CWebServer::AskForAuthentication(const HTTPRequest& request) const
...
   ret =
-      MHD_queue_basic_auth_fail_response(request.connection, CCompileInfo::GetAppName(), response);
+      (MHD_RESULT) MHD_queue_basic_auth_fail_response(request.connection, CCompileInfo::GetAppName(), response);
```
Comment (Rechi) [inline xbmc/network/WebServer.cpp:120]: > The `ColumnLimit` in `.clang-format` is set to `100` which defines line length [...]

https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md#3-formatting


> New code has to use C++ style casts and not older C style casts. [...]

https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md#122-casts

File: xbmc/network/WebServer.cpp [hunk 3]
```diff
@@ -172,7 +175,7 @@ int CWebServer::AnswerToConnection(void *cls, struct MHD_Connection *connection,
   return webServer->HandlePartialRequest(connection, connectionHandler, request, upload_data, upload_data_size, con_cls);
 }
 
-int CWebServer::HandlePartialRequest(struct MHD_Connection *connection, ConnectionHandler* connectionHandler, const HTTPRequest& request, const char *upload_data, size_t *upload_data_size, void **con_cls)
+MHD_RESULT CWebServer::HandlePartialRequest(struct MHD_Connection *connection, ConnectionHandler* connectionHandler, const HTTPRequest& request, const char *upload_data, size_t *upload_data_size, void **con_cls)
```
Comment (Rechi) [inline xbmc/network/WebServer.cpp:178]: > The `ColumnLimit` in `.clang-format` is set to `100` which defines line length [...]

https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md#3-formatting

File: xbmc/network/WebServer.cpp [hunk 4]
```diff
@@ -357,7 +360,7 @@ int CWebServer::HandleRequest(const std::shared_ptr<IHTTPRequestHandler>& handle
   return FinalizeRequest(handler, responseDetails.status, response);
 }
 
-int CWebServer::FinalizeRequest(const std::shared_ptr<IHTTPRequestHandler>& handler, int responseStatus, struct MHD_Response *response)
+MHD_RESULT CWebServer::FinalizeRequest(const std::shared_ptr<IHTTPRequestHandler>& handler, int responseStatus, struct MHD_Response *response)
```
Comment (Rechi) [inline xbmc/network/WebServer.cpp:363]: > The `ColumnLimit` in `.clang-format` is set to `100` which defines line length [...]

https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md#3-formatting

File: xbmc/network/WebServer.cpp [hunk 5]
```diff
@@ -574,7 +577,7 @@ void CWebServer::FinalizePostDataProcessing(ConnectionHandler *connectionHandler
   MHD_destroy_post_processor(connectionHandler->postprocessor);
 }
 
-int CWebServer::CreateMemoryDownloadResponse(const std::shared_ptr<IHTTPRequestHandler>& handler, struct MHD_Response *&response) const
+MHD_RESULT CWebServer::CreateMemoryDownloadResponse(const std::shared_ptr<IHTTPRequestHandler>& handler, struct MHD_Response *&response) const
```
Comment (Rechi) [inline xbmc/network/WebServer.cpp:580]: > The `ColumnLimit` in `.clang-format` is set to `100` which defines line length [...]

https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md#3-formatting

File: xbmc/network/WebServer.cpp [hunk 6]
```diff
@@ -635,7 +638,7 @@ int CWebServer::CreateMemoryDownloadResponse(const std::shared_ptr<IHTTPRequestH
   return CreateRangedMemoryDownloadResponse(handler, response);
 }
 
-int CWebServer::CreateRangedMemoryDownloadResponse(const std::shared_ptr<IHTTPRequestHandler>& handler, struct MHD_Response *&response) const
+MHD_RESULT CWebServer::CreateRangedMemoryDownloadResponse(const std::shared_ptr<IHTTPRequestHandler>& handler, struct MHD_Response *&response) const
```
Comment (Rechi) [inline xbmc/network/WebServer.cpp:641]: > The `ColumnLimit` in `.clang-format` is set to `100` which defines line length [...]

https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md#3-formatting

File: xbmc/network/WebServer.cpp [hunk 7]
```diff
@@ -716,7 +719,7 @@ int CWebServer::CreateRangedMemoryDownloadResponse(const std::shared_ptr<IHTTPRe
   return CreateMemoryDownloadResponse(request.connection, result.c_str(), result.size(), false, true, response);
 }
 
-int CWebServer::CreateRedirect(struct MHD_Connection *connection, const std::string &strURL, struct MHD_Response *&response) const
+MHD_RESULT CWebServer::CreateRedirect(struct MHD_Connection *connection, const std::string &strURL, struct MHD_Response *&response) const
```
Comment (Rechi) [inline xbmc/network/WebServer.cpp:722]: > The `ColumnLimit` in `.clang-format` is set to `100` which defines line length [...]

https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md#3-formatting

File: xbmc/network/WebServer.cpp [hunk 8]
```diff
@@ -729,7 +732,7 @@ int CWebServer::CreateRedirect(struct MHD_Connection *connection, const std::str
   return MHD_YES;
 }
 
-int CWebServer::CreateFileDownloadResponse(const std::shared_ptr<IHTTPRequestHandler>& handler, struct MHD_Response *&response) const
+MHD_RESULT CWebServer::CreateFileDownloadResponse(const std::shared_ptr<IHTTPRequestHandler>& handler, struct MHD_Response *&response) const
```
Comment (Rechi) [inline xbmc/network/WebServer.cpp:735]: > The `ColumnLimit` in `.clang-format` is set to `100` which defines line length [...]

https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md#3-formatting

File: xbmc/network/WebServer.cpp [hunk 9]
```diff
@@ -868,7 +871,7 @@ int CWebServer::CreateFileDownloadResponse(const std::shared_ptr<IHTTPRequestHan
   return MHD_YES;
 }
 
-int CWebServer::CreateErrorResponse(struct MHD_Connection *connection, int responseType, HTTPMethod method, struct MHD_Response *&response) const
+MHD_RESULT CWebServer::CreateErrorResponse(struct MHD_Connection *connection, int responseType, HTTPMethod method, struct MHD_Response *&response) const
```
Comment (Rechi) [inline xbmc/network/WebServer.cpp:874]: > The `ColumnLimit` in `.clang-format` is set to `100` which defines line length [...]

https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md#3-formatting

File: xbmc/network/WebServer.cpp [hunk 10]
```diff
@@ -899,7 +902,7 @@ int CWebServer::CreateErrorResponse(struct MHD_Connection *connection, int respo
   return MHD_YES;
 }
 
-int CWebServer::CreateMemoryDownloadResponse(struct MHD_Connection *connection, const void *data, size_t size, bool free, bool copy, struct MHD_Response *&response) const
+MHD_RESULT CWebServer::CreateMemoryDownloadResponse(struct MHD_Connection *connection, const void *data, size_t size, bool free, bool copy, struct MHD_Response *&response) const
```
Comment (Rechi) [inline xbmc/network/WebServer.cpp:905]: > The `ColumnLimit` in `.clang-format` is set to `100` which defines line length [...]

https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md#3-formatting

File: xbmc/network/WebServer.cpp [hunk 11]
```diff
@@ -911,20 +914,20 @@ int CWebServer::CreateMemoryDownloadResponse(struct MHD_Connection *connection,
   return MHD_YES;
 }
 
-int CWebServer::SendResponse(const HTTPRequest& request, int responseStatus, MHD_Response *response) const
+MHD_RESULT CWebServer::SendResponse(const HTTPRequest& request, int responseStatus, MHD_Response *response) const
```
Comment (Rechi) [inline xbmc/network/WebServer.cpp:917]: > The `ColumnLimit` in `.clang-format` is set to `100` which defines line length [...]

https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md#3-formatting

File: xbmc/network/WebServer.cpp [hunk 12]
```diff
@@ -911,20 +914,20 @@ int CWebServer::CreateMemoryDownloadResponse(struct MHD_Connection *connection,
...
 
-  int ret = MHD_queue_response(request.connection, responseStatus, response);
+  MHD_RESULT ret = MHD_queue_response(request.connection, responseStatus, response);
   MHD_destroy_response(response);
 
   return ret;
 }
 
-int CWebServer::SendErrorResponse(const HTTPRequest& request, int errorType, HTTPMethod method) const
+MHD_RESULT CWebServer::SendErrorResponse(const HTTPRequest& request, int errorType, HTTPMethod method) const
```
Comment (Rechi) [inline xbmc/network/WebServer.cpp:927]: > The `ColumnLimit` in `.clang-format` is set to `100` which defines line length [...]

https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md#3-formatting

File: xbmc/network/WebServer.cpp [hunk 13]
```diff
@@ -1331,10 +1334,10 @@ std::string CWebServer::CreateMimeTypeFromExtension(const char *ext)
   return CMime::GetMimeType(ext);
 }
 
-int CWebServer::AddHeader(struct MHD_Response *response, const std::string &name, const std::string &value) const
+MHD_RESULT CWebServer::AddHeader(struct MHD_Response *response, const std::string &name, const std::string &value) const
```
Comment (Rechi) [inline xbmc/network/WebServer.cpp:1337]: > The `ColumnLimit` in `.clang-format` is set to `100` which defines line length [...]

https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md#3-formatting

File: xbmc/network/WebServer.h [hunk 14]
```diff
@@ -57,17 +57,17 @@ class CWebServer
 
   virtual void LogRequest(const char* uri) const;
 
-  virtual int HandlePartialRequest(struct MHD_Connection *connection, ConnectionHandler* connectionHandler, const HTTPRequest& request,
+  virtual MHD_RESULT HandlePartialRequest(struct MHD_Connection *connection, ConnectionHandler* connectionHandler, const HTTPRequest& request,
```
Comment (Rechi) [inline xbmc/network/WebServer.h:60]: > The `ColumnLimit` in `.clang-format` is set to `100` which defines line length [...]

https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md#3-formatting

File: xbmc/network/WebServer.h [hunk 15]
```diff
@@ -57,17 +57,17 @@ class CWebServer
...
                                    const char *upload_data, size_t *upload_data_size, void **con_cls);
-  virtual int HandleRequest(const std::shared_ptr<IHTTPRequestHandler>& handler);
-  virtual int FinalizeRequest(const std::shared_ptr<IHTTPRequestHandler>& handler, int responseStatus, struct MHD_Response *response);
+  virtual MHD_RESULT HandleRequest(const std::shared_ptr<IHTTPRequestHandler>& handler);
+  virtual MHD_RESULT FinalizeRequest(const std::shared_ptr<IHTTPRequestHandler>& handler, int responseStatus, struct MHD_Response *response);
```
Comment (Rechi) [inline xbmc/network/WebServer.h:63]: > The `ColumnLimit` in `.clang-format` is set to `100` which defines line length [...]

https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md#3-formatting

File: xbmc/network/WebServer.h [hunk 16]
```diff
@@ -77,18 +77,18 @@ class CWebServer
   bool ProcessPostData(const HTTPRequest& request, ConnectionHandler *connectionHandler, const char *upload_data, size_t *upload_data_size, void **con_cls) const;
   void FinalizePostDataProcessing(ConnectionHandler *connectionHandler) const;
 
-  int CreateMemoryDownloadResponse(const std::shared_ptr<IHTTPRequestHandler>& handler, struct MHD_Response *&response) const;
-  int CreateRangedMemoryDownloadResponse(const std::shared_ptr<IHTTPRequestHandler>& handler, struct MHD_Response *&response) const;
+  MHD_RESULT CreateMemoryDownloadResponse(const std::shared_ptr<IHTTPRequestHandler>& handler, struct MHD_Response *&response) const;
```
Comment (Rechi) [inline xbmc/network/WebServer.h:80]: > The `ColumnLimit` in `.clang-format` is set to `100` which defines line length [...]

https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md#3-formatting

File: xbmc/network/WebServer.h [hunk 17]
```diff
@@ -77,18 +77,18 @@ class CWebServer
...
+  MHD_RESULT CreateRangedMemoryDownloadResponse(const std::shared_ptr<IHTTPRequestHandler>& handler, struct MHD_Response *&response) const;
```
Comment (Rechi) [inline xbmc/network/WebServer.h:81]: > The `ColumnLimit` in `.clang-format` is set to `100` which defines line length [...]

https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md#3-formatting

File: xbmc/network/WebServer.h [hunk 18]
```diff
@@ -77,18 +77,18 @@ class CWebServer
...
 
-  int CreateRedirect(struct MHD_Connection *connection, const std::string &strURL, struct MHD_Response *&response) const;
-  int CreateFileDownloadResponse(const std::shared_ptr<IHTTPRequestHandler>& handler, struct MHD_Response *&response) const;
-  int CreateErrorResponse(struct MHD_Connection *connection, int responseType, HTTPMethod method, struct MHD_Response *&response) const;
-  int CreateMemoryDownloadResponse(struct MHD_Connection *connection, const void *data, size_t size, bool free, bool copy, struct MHD_Response *&response) const;
+  MHD_RESULT CreateRedirect(struct MHD_Connection *connection, const std::string &strURL, struct MHD_Response *&response) const;
```
Comment (Rechi) [inline xbmc/network/WebServer.h:83]: > The `ColumnLimit` in `.clang-format` is set to `100` which defines line length [...]

https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md#3-formatting

File: xbmc/network/WebServer.h [hunk 19]
```diff
@@ -77,18 +77,18 @@ class CWebServer
...
+  MHD_RESULT CreateFileDownloadResponse(const std::shared_ptr<IHTTPRequestHandler>& handler, struct MHD_Response *&response) const;
```
Comment (Rechi) [inline xbmc/network/WebServer.h:84]: > The `ColumnLimit` in `.clang-format` is set to `100` which defines line length [...]

https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md#3-formatting

File: xbmc/network/WebServer.h [hunk 20]
```diff
@@ -77,18 +77,18 @@ class CWebServer
...
+  MHD_RESULT CreateErrorResponse(struct MHD_Connection *connection, int responseType, HTTPMethod method, struct MHD_Response *&response) const;
```
Comment (Rechi) [inline xbmc/network/WebServer.h:85]: > The `ColumnLimit` in `.clang-format` is set to `100` which defines line length [...]

https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md#3-formatting

File: xbmc/network/WebServer.h [hunk 21]
```diff
@@ -77,18 +77,18 @@ class CWebServer
...
+  MHD_RESULT CreateMemoryDownloadResponse(struct MHD_Connection *connection, const void *data, size_t size, bool free, bool copy, struct MHD_Response *&response) const;
```
Comment (Rechi) [inline xbmc/network/WebServer.h:86]: > The `ColumnLimit` in `.clang-format` is set to `100` which defines line length [...]

https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md#3-formatting

File: xbmc/network/WebServer.h [hunk 22]
```diff
@@ -77,18 +77,18 @@ class CWebServer
...
 
-  int SendResponse(const HTTPRequest& request, int responseStatus, MHD_Response *response) const;
-  int SendErrorResponse(const HTTPRequest& request, int errorType, HTTPMethod method) const;
+  MHD_RESULT SendResponse(const HTTPRequest& request, int responseStatus, MHD_Response *response) const;
```
Comment (Rechi) [inline xbmc/network/WebServer.h:88]: > The `ColumnLimit` in `.clang-format` is set to `100` which defines line length [...]

https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md#3-formatting

File: xbmc/network/WebServer.h [hunk 23]
```diff
@@ -77,18 +77,18 @@ class CWebServer
...
+  MHD_RESULT SendErrorResponse(const HTTPRequest& request, int errorType, HTTPMethod method) const;
 
-  int AddHeader(struct MHD_Response *response, const std::string &name, const std::string &value) const;
+  MHD_RESULT AddHeader(struct MHD_Response *response, const std::string &name, const std::string &value) const;
```
Comment (Rechi) [inline xbmc/network/WebServer.h:91]: > The `ColumnLimit` in `.clang-format` is set to `100` which defines line length [...]

https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md#3-formatting

File: xbmc/network/httprequesthandler/HTTPRequestHandlerUtils.cpp [hunk 24]
```diff
@@ -62,7 +62,7 @@ bool HTTPRequestHandlerUtils::GetRequestedRanges(struct MHD_Connection *connecti
   return ranges.Parse(GetRequestHeaderValue(connection, MHD_HEADER_KIND, MHD_HTTP_HEADER_RANGE), totalLength);
 }
 
-int HTTPRequestHandlerUtils::FillArgumentMap(void *cls, enum MHD_ValueKind kind, const char *key, const char *value)
+MHD_RESULT HTTPRequestHandlerUtils::FillArgumentMap(void *cls, enum MHD_ValueKind kind, const char *key, const char *value)
```
Comment (Rechi) [inline xbmc/network/httprequesthandler/HTTPRequestHandlerUtils.cpp:65]: > The `ColumnLimit` in `.clang-format` is set to `100` which defines line length [...]

https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md#3-formatting

File: xbmc/network/httprequesthandler/HTTPRequestHandlerUtils.cpp [hunk 25]
```diff
@@ -73,7 +73,7 @@ int HTTPRequestHandlerUtils::FillArgumentMap(void *cls, enum MHD_ValueKind kind,
   return MHD_YES;
 }
 
-int HTTPRequestHandlerUtils::FillArgumentMultiMap(void *cls, enum MHD_ValueKind kind, const char *key, const char *value)
+MHD_RESULT HTTPRequestHandlerUtils::FillArgumentMultiMap(void *cls, enum MHD_ValueKind kind, const char *key, const char *value)
```
Comment (Rechi) [inline xbmc/network/httprequesthandler/HTTPRequestHandlerUtils.cpp:76]: > The `ColumnLimit` in `.clang-format` is set to `100` which defines line length [...]

https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md#3-formatting

File: xbmc/network/httprequesthandler/HTTPRequestHandlerUtils.h [hunk 26]
```diff
@@ -25,6 +25,6 @@ class HTTPRequestHandlerUtils
 private:
   HTTPRequestHandlerUtils() = delete;
 
-  static int FillArgumentMap(void *cls, enum MHD_ValueKind kind, const char *key, const char *value);
-  static int FillArgumentMultiMap(void *cls, enum MHD_ValueKind kind, const char *key, const char *value);
+  static MHD_RESULT FillArgumentMap(void *cls, enum MHD_ValueKind kind, const char *key, const char *value);
```
Comment (Rechi) [inline xbmc/network/httprequesthandler/HTTPRequestHandlerUtils.h:28]: > The `ColumnLimit` in `.clang-format` is set to `100` which defines line length [...]

https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md#3-formatting

File: xbmc/network/httprequesthandler/HTTPRequestHandlerUtils.h [hunk 27]
```diff
@@ -25,6 +25,6 @@ class HTTPRequestHandlerUtils
...
+  static MHD_RESULT FillArgumentMultiMap(void *cls, enum MHD_ValueKind kind, const char *key, const char *value);
```
Comment (Rechi) [inline xbmc/network/httprequesthandler/HTTPRequestHandlerUtils.h:29]: > The `ColumnLimit` in `.clang-format` is set to `100` which defines line length [...]

https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md#3-formatting

File: xbmc/network/httprequesthandler/IHTTPRequestHandler.h [hunk 28]
```diff
@@ -22,6 +22,14 @@
 #include <sys/socket.h>
 #include <sys/types.h>
 
+#include <microhttpd.h>
```
Comment (Rechi) [inline xbmc/network/httprequesthandler/IHTTPRequestHandler.h:25]: This include already exist at the correct place a few lines above.

**Review by Rechi — COMMENTED** [review]
Please follow the [current code guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md).

File: xbmc/network/httprequesthandler/IHTTPRequestHandler.h (hunk 28 above)
Comment (candrews) [reply xbmc/network/httprequesthandler/IHTTPRequestHandler.h:25]: Thanks!

File: xbmc/network/httprequesthandler/IHTTPRequestHandler.h [hunk 29]
```diff
@@ -22,6 +22,12 @@
 #include <sys/socket.h>
 #include <sys/types.h>
 
+#if MHD_VERSION >= 0x00097002
+#define MHD_RESULT enum MHD_Result
+#else
+#define MHD_RESULT int
+#endif
+
```
Comment (yol) [inline xbmc/network/httprequesthandler/IHTTPRequestHandler.h:30]: Would this work?

```
#if MHD_VERSION >= 0x00097002
using MHD_RESULT = MHD_Result;
#else
using MHD_RESULT = int;
#endif
```

If yes, I would much prefer it.

Comment (candrews) [reply xbmc/network/httprequesthandler/IHTTPRequestHandler.h:30]: I tested that and it worked fine. I've added a commit to this PR with that change.

File: xbmc/network/WebServer.cpp [hunk 30]
```diff
@@ -113,8 +117,12 @@ int CWebServer::AskForAuthentication(const HTTPRequest& request) const
 
   LogResponse(request, MHD_HTTP_UNAUTHORIZED);
 
-  ret =
-      MHD_queue_basic_auth_fail_response(request.connection, CCompileInfo::GetAppName(), response);
+  // This MHD_RESULT cast is only necessary for libmicrohttpd 0.9.71
+  // The return type of MHD_queue_basic_auth_fail_response was fixed for future versions
+  // See
+  // https://git.gnunet.org/libmicrohttpd.git/commit/?id=860b42e9180da4dcd7e8690a3fcdb4e37e5772c5
+  ret = (MHD_RESULT)MHD_queue_basic_auth_fail_response(request.connection,
```
Comment (yol) [inline xbmc/network/WebServer.cpp:124]: This is still a C-style cast, please change

**Review by yol — CHANGES_REQUESTED** [review]
Squash b425d3c into d0c0823 and fix the C-style cast, then good to merge from my side :+1:

File: xbmc/network/WebServer.cpp (hunk 30 above)
Comment (candrews) [reply xbmc/network/WebServer.cpp]: Change to `static_cast<MHD_RESULT>(MHD_queue_basic_auth_fail_response(request.connection, CCompileInfo::GetAppName(), response));`

**Comment by candrews** [comment]
> Squash [b425d3c](https://github.com/xbmc/xbmc/commit/b425d3c0458b4acf08bcb315c5a1e5a5e75cb816) into [d0c0823](https://github.com/xbmc/xbmc/commit/d0c08238eee5f2bad901df41800a9999ca7df156) and fix the C-style cast, then good to merge from my side +1

Done!

**Review by yol — APPROVED** [review]

**Comment by yol** [comment]
jenkins build and merge

**Comment by yol** [comment]
let's try again :sunglasses:

**Comment by yol** [comment]
jenkins build and merge

**Pull request merged**